frontend/src/metabase/static-viz/components/XYChart/XYChart.tsx

```diff
@@ -202,20 +219,49 @@ export const XYChart = ({
       />
 
       <Group left={valuesLeftOffset}>
+        <Group top={margin.top} left={xMin}>
+          {defaultYScale && (
+            <GridRows
+              scale={defaultYScale}
+              width={calculatedInnerWidth}
+              strokeDasharray="4"
+            />
+          )}
+        </Group>
+
         <AxisBottom
           scale={xScale.scale}
-          label={areXTicksRotated ? undefined : settings.labels.bottom}
+          label={settings.labels.bottom}
           top={yMin}
           left={xMin}
           numTicks={xTicksCount}
-          labelOffset={xLabelOffset}
+          labelOffset={getXLabelOffset(
+            xTickDisplay,
+            style.axes.ticks.fontSize,
+            Math.max(
```

Comment:
You should not calculate offset here based on ticks dimensions. We calculate margins based on these ticks dimensions so that `yMin` and `margin.bottom` should be enough to place this label

Reply:
I agreed, it was hard to understand the whole picture, but the reason I do this in the first place was because I thought ticks dimensions weren't accounted for as I found an instnace that the Y label could protrude the legend
![image](https://user-images.githubusercontent.com/1937582/188825556-274bc854-5181-490e-aabf-85677e8509e0.png)

That's fair given hor the label was just rotated 45deg compared to 90deg now, but I should've fixed this maybe in the `calculateMargin` instead
https://github.com/metabase/metabase/blob/93d3df58137aebe7d041a4e16325c1d50f6efd55/frontend/src/metabase/static-viz/components/XYChart/utils/margin.ts#L23

Anyway, I'll pick this up again, after I finish other more important regressions and low-hanging fruit items.